submariner/IPSec-transport-mode.md

```diff
@@ -0,0 +1,132 @@
+[root@cluster1-worker]# ipsec whack --psk --encrypt --forceencaps --name submariner-cable-cluster2-172-18-0-7-1 --host 172.18.0.8 --clientprotoport udp --ikeport 4500 --to --host 172.18.0.7 --clientprotoport udp/vxlan --ikeport 4500
+[root@cluster1-worker]# ipsec whack --route --name submariner-cable-cluster2-172-18-0-7-0
+[root@cluster1-worker]# ipsec whack --route --name submariner-cable-cluster2-172-18-0-7-1
+[root@cluster1-worker]# ipsec whack --initiate --asynchronous --name submariner-cable-cluster2-172-18-0-7-0
+[root@cluster1-worker]# ipsec whack --initiate --asynchronous --name submariner-cable-cluster2-172-18-0-7-1
+```
+
+```console
+[root@cluster2-worker]# ipsec whack --psk --encrypt --forceencaps --name submariner-cable-cluster1-172-18-0-8-0 --host 172.18.0.7 --clientprotoport udp/vxlan --ikeport 4500 --to --host 172.18.0.8 --clientprotoport udp --ikeport 4500
+[root@cluster2-worker]# ipsec whack --psk --encrypt --forceencaps --name submariner-cable-cluster1-172-18-0-8-1 --host 172.18.0.7 --clientprotoport udp --ikeport 4500 --to --host 172.18.0.8 --clientprotoport udp/vxlan --ikeport 4500
+[root@cluster2-worker]# ipsec whack --route --name submariner-cable-cluster1-172-18-0-8-0
+[root@cluster2-worker]# ipsec whack --route --name submariner-cable-cluster1-172-18-0-8-1
+[root@cluster2-worker]# ipsec whack --initiate --asynchronous --name submariner-cable-cluster1-172-18-0-8-0
+[root@cluster2-worker]# ipsec whack --initiate --asynchronous --name submariner-cable-cluster1-172-18-0-8-1
+```
+<!-- markdownlint-enable line-length -->
+
+> **_NOTE:_** The clientprotoport parameter is used to specify the Port Selectors (filters) to be used on this connection.
+> The general form is protocol/port.
+
+The resulting topology is shown below
+![Submariner VXLAN over IPSec transport mode cable driver view](./images/vxlan-over-ipsec-transport.png)
+
+### Hardware offload enablement
+
+_"libreswan as of version 3.23 supports the new cryptographic hardware offload as implemented by Linux 4.11 and up using the
+native (XFRM) IPsec stack. Libreswan autodetects supports for any hardware supporting this crypto offload API"_ [1].
+
+The [NICs](https://libreswan.org/wiki/Cryptographic_Acceleration#Supported_hardware) that support Hardware offload capabilities
+for Libreswan are pretty standard and available in all cloud environments. Submariner encourages the 'selection' of one these
+NICs for a node where one intends to deploy the Submariner gateway and leverage the use of labels to schedule the pod to the
+appropriate node: a node label for the node that has the right NIC and for the GW pod to use a label selector (for that node label)
+to ensure that the gateway lands on the right platform. Then rely on Libreswan to autodetect the HW features and take advantage of them.
+
+> **_NOTE:_** The IPSec cable driver will need to be modified to run `ipsec _stackmanager start` to load the appropriate
```

Comment:
Thanks for adding the HW offload section  @maryamtahhan .

IIUC,` ipsec _stackmanager start` should run at node level and should affect all ipsec consumers on this node.

Maybe it makes sense to run this operation in some other component outside SM ? because SM GW pod might be only one of the ipsec HW offload consumers  on this node

So, to allow ipsec HW offload :

A. some other component should run all the bits to enable ipsec HW offload and set the relevant label for the appropriate nodes

B. Guide K8S to schedule SM gateway pods on these nodes